pkg/operator/controller/controller.go

```diff
@@ -200,7 +177,63 @@ func (r *reconciler) reconcile() (reconcile.Result, error) {
 		}
 	}
 
-	return reconcile.Result{}, utilerrors.NewAggregate(errs)
+	return result, utilerrors.NewAggregate(errs)
+}
+
+// updateConfig updates DNS and Infra config
+func (r *reconciler) updateConfig(result *reconcile.Result) error {
```

Comment:
I think just doing an inline get for these and passing them to functions which need them more clear — this refactor 

1. Adds unnecessary state to the controller, and
2. Obscures the resource dependencies by introducing ad-hoc reference to controller state

Reply:
done